examples/MQ/pixelSimSplit/src/devices/FairMQChunkMerger.cxx

```diff
@@ -120,21 +119,23 @@ bool FairMQChunkMerger::MergeData(fair::mq::Parts& parts, int /*index*/)
                 continue;   //  want only MCTrack array to renumber tracks and get track shifts...
             //  LOG(info) << "BEFORE ADDING, TCA \"" << tca_elem->GetName() << "\" has " <<
             //  tca_elem->GetEntries() << " entries.";
-            TClonesArray* arrayToAdd;
             for (auto& it = fRet.first; it != fRet.second; ++it) {
                 if (it->second.first == fMCSplitEventHeader->GetChunkStart())
                     continue;
                 if (strcmp(tca_elem->GetName(), it->second.second->GetName()) == 0) {
                     trackShift.push_back(tca_elem->GetEntries());
-                    arrayToAdd = dynamic_cast<TClonesArray*>(it->second.second);
+                    TClonesArray* arrayToAdd = dynamic_cast<TClonesArray*>(it->second.second);
                     for (int iobj = 0; iobj < arrayToAdd->GetEntries(); ++iobj) {
```

Comment:
Taking a fresh look at this code, already the `arrayToAdd->GetEntries()` (in the for loop) requires `arrayToAdd != nullptr`.
So testing just before the `->AbsorbObjects()` call below is a good thing! But maybe too late?

What about something like this?

```suggestion
                    auto arrayToAdd = dynamic_cast<TClonesArray*>(it->second.second);
                    if (!arrayToAdd) {
                        LOG(warning) << "Did not find TCA with name \"" << tca_elem->GetName() << "\"";
                        continue;
                    }
                    for (int iobj = 0; iobj < arrayToAdd->GetEntries(); ++iobj) {
```